hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/spi/state/impl/RebuiltStateImpl.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi.state.impl;
+
+import com.hedera.node.app.spi.state.State;
+import com.hedera.node.app.spi.state.StateKey;
+import com.swirlds.fchashmap.FCHashMap;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+
+/**
+ * An implementation of {@link State} backed by a {@link FCHashMap} for aliases, that needs to be
+ * rebuilt from the {@link com.hedera.services.ServicesState}.
+ *
+ * @param <K> The type of key for the state
+ * @param <V> The type of value for the state
+ */
+public class RebuiltStateImpl<K, V> extends StateBase<K, V> {
```

Comment:
What is the purpose of this class? Why do we need an implementation based on rebuilt state different from genesis state?

Reply:
Since this is a **FcHashMap** of `aliases` that are rebuilt every time we load state. If we wanted to not have a separate implementation , could we move the aliases map to be serialized to state ?